packages/widget/src/components/resource-amount-selector/resource-amount-selector.ts

```diff
@@ -43,6 +41,16 @@ export class AmountSelector extends BaseComponent {
   @state() validationMessage: string | null = null;
   @state() amount: string = '0';
 
+  constructor() {
+    super();
+    document.addEventListener('wallet-disconnected', () => {
```

Comment:
instead of doing this everywhere, can you move this logic into transferController?

Reply:
good idea! I tried this approach but from the first try it didn't go well :) Please check the updated code